src/user.py

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env python3
+
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""User class."""
+
+import logging
+import secrets
+import string
+
+# pydantic is causing this no-name-in-module problem
+from pydantic import BaseModel, Field, validator  # pylint: disable=no-name-in-module,import-error
+
+logger = logging.getLogger(__name__)
+
+
+def _generate_password() -> str:
+    """Set password to user. Extracted from postgresql-k8s charm.
+
+    Returns:
+        random password.
+    """
+    choices = string.ascii_letters + string.digits
+    password = "".join([secrets.choice(choices) for i in range(16)])
```

Comment:
```suggestion
    password = "".join(secrets.choice(choices) for _ in range(16))
```
The list wrapper can be removed since join takes an iterable :D

Reply:
Related: #52